rpc/wrtc_call_queue_memory.go

```diff
@@ -41,42 +38,27 @@ func newMemoryWebRTCCallQueueTest(logger utils.ZapCompatibleLogger) *memoryWebRT
 }
 
 func newMemoryWebRTCCallQueue(uuidDeterministic bool, logger utils.ZapCompatibleLogger) *memoryWebRTCCallQueue {
-	cancelCtx, cancelFunc := context.WithCancel(context.Background())
 	queue := &memoryWebRTCCallQueue{
 		hostQueues:        map[string]*singleWebRTCHostQueue{},
-		cancelCtx:         cancelCtx,
-		cancelFunc:        cancelFunc,
 		uuidDeterministic: uuidDeterministic,
 		logger:            logger,
 	}
-	queue.activeBackgroundWorkers.Add(1)
-	ticker := time.NewTicker(5 * time.Second)
-	utils.ManagedGo(func() {
-		for {
-			if cancelCtx.Err() != nil {
-				return
-			}
-			select {
-			case <-cancelCtx.Done():
-				return
-			case <-ticker.C:
-			}
-			now := time.Now()
-			queue.mu.Lock()
-			for _, hostQueue := range queue.hostQueues {
-				hostQueue.mu.Lock()
-				for offerID, offer := range hostQueue.activeOffers {
-					if d, ok := offer.offer.answererDoneCtx.Deadline(); ok && d.Before(now) {
-						delete(hostQueue.activeOffers, offerID)
-					}
+	queue.activeBackgroundWorkers = utils.NewStoppableWorkerWithTicker(5*time.Second, func(ctx context.Context) {
+		if ctx.Err() != nil {
```

Comment:
Before, this was necessary to exit the loop. But this is no longer serving a purpose. The stoppable workers caller checks the state/context before calling this method.